### ISSUE TITLE
DHT11 library doesn't work

### ISSUE THREAD

**Issue by ljb7977** [jguille2/SA5#6]
Hello,

I've used the DHT11 block of your library well, but it suddenly doesn't work now. When I click the block, it doesn't report any value and stays clicked. I guess that the waiting block in DHT block doesn't end, but I'm not sure. I tried to fix it by myself but I can't figure out what the problem is...
Could you check the problem out?

Thank you.

**Comment by jguille2** [comment]
Hi @ljb7977 ,

I was working in new release (2.3) for the SA5Firmata that it will run also in Snap4Arduino online (chromium version). The work is done, but it will be released with the next version of Snap4Arduino (it needs some changes in the core). The good news is that these blocks will be into a common library into the program... and then, their implementation will be easier.

Testing the new blocks, I saw a problem with DHT11. It was strange because I saw the problem was with a DHT11 module... and if I tested with other (the one I had used when I made the blocks) the block runs ok. So I made some changes into the DHT11 library (changing some times between the readings) and then both modules ran ok.

If you want to test, I post here a "prototype" (I've made some changes to run with current Snap4Arduino version). The zip has a new firmware (ino) to flush the board and a xml with the modified block.
[dht11testing23.zip](https://github.com/jguille2/SA5/files/2182443/dht11testing23.zip)

Joan
